crates/webidl/src/lib.rs

```diff
@@ -453,61 +453,61 @@ impl<'src> FirstPassRecord<'src> {
-        // always throw.
-        //
-        // From https://heycam.github.io/webidl/#Constructor (emphasis
-        // mine):
-        //
-        // > The prose definition of a constructor must either return an IDL
-        // > value of a type corresponding to the interface the
-        // > `[Constructor]` extended attribute appears on, **or throw an
-        // > exception**.
-        let throws = true;
-
-        for import_function in self.create_function(
-            "new",
-            overloaded,
-            same_argument_names,
-            &match self.convert_arguments(args) {
-                Some(arguments) => arguments,
-                None => return,
-            },
-            IdlType::Interface(iface_name),
-            kind,
-            structural,
-            throws,
-            None,
-        ) {
-            program.imports.push(wrap_import_function(import_function));
-        }
-    }
+    // fn append_constructor(
+    //     &self,
+    //     program: &mut backend::ast::Program,
+    //     iface_name: &'src str,
+    //     ctor_name: &'src str,
+    //     args: &[Argument<'src>],
+    // ) {
+    //     let (overloaded, same_argument_names) = self.get_operation_overloading(
+    //         args,
+    //         &::first_pass::OperationId::Constructor,
+    //         iface_name,
+    //         false,
+    //     );
+    //
+    //     let self_ty = ident_ty(rust_ident(camel_case_ident(iface_name).as_str()));
+    //
+    //     let kind = backend::ast::ImportFunctionKind::Method {
+    //         class: ctor_name.to_string(),
+    //         ty: self_ty.clone(),
+    //         kind: backend::ast::MethodKind::Constructor,
+    //     };
+    //
+    //     let structural = false;
+    //
+    //     // Constructors aren't annotated with `[Throws]` extended attributes
+    //     // (how could they be, since they themselves are extended
+    //     // attributes?) so we must conservatively assume that they can
+    //     // always throw.
+    //     //
+    //     // From https://heycam.github.io/webidl/#Constructor (emphasis
+    //     // mine):
+    //     //
+    //     // > The prose definition of a constructor must either return an IDL
+    //     // > value of a type corresponding to the interface the
+    //     // > `[Constructor]` extended attribute appears on, **or throw an
+    //     // > exception**.
+    //     let throws = true;
+    //
+    //     for import_function in self.create_function(
+    //         "new",
+    //         overloaded,
+    //         same_argument_names,
+    //         &match self.convert_arguments(args) {
+    //             Some(arguments) => arguments,
+    //             None => return,
+    //         },
+    //         IdlType::Interface(iface_name),
+    //         kind,
+    //         structural,
+    //         throws,
+    //         None,
+    //     ) {
+    //         program.imports.push(wrap_import_function(import_function));
+    //     }
+    // }
```

Comment:
I'm assuming all this will be removed in follow up commits?

Reply:
indeed!